warp/cairo-src/evm/uint256.cairo

```diff
@@ -108,13 +108,27 @@ func sgt{range_check_ptr}(op1 : Uint256, op2 : Uint256) -> (result : Uint256):
 end
 
 func uint256_mod{range_check_ptr}(a : Uint256, m : Uint256) -> (res : Uint256):
-    assert 0 = 1  # not implemented yet =(
-    jmp rel 0
+    let (early) = uint256_lt(a, m)
+    if early == 1:
+        tempvar range_check_ptr = range_check_ptr
+        return (res=a)
+    else:
+        tempvar range_check_ptr = range_check_ptr
+        let (_, rem : Uint256) = uint256_unsigned_div_rem(a, m)
+        return (res=rem)
+    end
 end
 
 func uint256_addmod{range_check_ptr}(a : Uint256, b : Uint256, m : Uint256) -> (res : Uint256):
-    assert 0 = 1  # not implemented yet =(
-    jmp rel 0
+    let (pre_mod : Uint256) = uint256_add(a, b)
+    let (res : Uint256) = uint256_mod(pre_mod, m)
+    return (res=res)
```

Comment:
The implementation of `addmod` and `mulmod` is not correct. Intermediate calculations, per yellow paper, should not be subject to modulo

The counterexample: `mulmod(2**128, 2**128, 3)` should be `3`, but in this implementation it is `0`, in fact, for every `m`.